### ISSUE TITLE
Getting `EOFError: Unexpected EOF while reading bytes` when reading data from clickhouse

### ISSUE THREAD

**Issue by kartikeyporwal** [mymarilyn/clickhouse-driver#397]
**Describe the bug**

Getting the below mentioned error traceback when retrieving data

```
  File "/usr/local/lib/python3.10/site-packages/clickhouse_driver/client.py", line 213, in iter_receive_result
    for rows in result:
  File "/usr/local/lib/python3.10/site-packages/clickhouse_driver/result.py", line 114, in __next__
    packet = next(self.packet_generator)
  File "/usr/local/lib/python3.10/site-packages/clickhouse_driver/client.py", line 220, in packet_generator
    packet = self.receive_packet()
  File "/usr/local/lib/python3.10/site-packages/clickhouse_driver/client.py", line 234, in receive_packet
    packet = self.connection.receive_packet()
  File "/usr/local/lib/python3.10/site-packages/clickhouse_driver/connection.py", line 565, in receive_packet
    packet.block = self.receive_data(may_be_use_numpy=True)
  File "/usr/local/lib/python3.10/site-packages/clickhouse_driver/connection.py", line 640, in receive_data
    return reader.read(use_numpy=use_numpy)
  File "/usr/local/lib/python3.10/site-packages/clickhouse_driver/streams/native.py", line 83, in read
    column = read_column(
  File "/usr/local/lib/python3.10/site-packages/clickhouse_driver/columns/service.py", line 154, in read_column
    return col.read_data(n_items, buf)
  File "/usr/local/lib/python3.10/site-packages/clickhouse_driver/columns/base.py", line 162, in read_data
    items = self._read_data(n_items, buf, nulls_map=nulls_map)
  File "/usr/local/lib/python3.10/site-packages/clickhouse_driver/columns/base.py", line 166, in _read_data
    items = self.read_items(n_items, buf)
  File "/usr/local/lib/python3.10/site-packages/clickhouse_driver/columns/base.py", line 210, in read_items
    return s.unpack(buf.read(s.size))
  File "clickhouse_driver/bufferedreader.pyx", line 40, in clickhouse_driver.bufferedreader.BufferedReader.read
  File "clickhouse_driver/bufferedreader.pyx", line 240, in clickhouse_driver.bufferedreader.BufferedSocketReader.read_into_buffer
EOFError: Unexpected EOF while reading bytes

```

**To Reproduce**
I'm not sure how it can be reproduced, because there is a certain data that it retrieves and then leads to this error.

**Expected behavior**
Error should not have occurred when retrieving data from clickhouse db

**Versions**

- Clickhouse Driver version: `0.2.6`
- ClickHouse server version: `22.1.3.7`
- Python version: `3.10.4`

https://github.com/mymarilyn/clickhouse-driver/issues/352 didn't work for me.

**Comment by Mew97** [comment]
I encountered the same error, but when I used the DB API to query, there was no such problem

**Versions**

- Clickhouse Driver version: 0.2.6
- ClickHouse server version: 23.11.1.2711
- Python version: 3.9.6

**Comment by kingsabru** [comment]
I also encountered this error. The problem was with me connecting to the host over HTTPS (or SSL) and me "not using the correct port" when creating the clickhouse-driver Client object. 

The original connection credentials I was given specified port 8443. However, the clickhouse-driver [Connection docs](https://clickhouse-driver.readthedocs.io/en/latest/api.html#connection) mandated using port **9440**. I also had to set `secure=True` (this is False by default).

In summary, the Client object definition should look like this:
```
from clickhouse_driver import Client

# WARNING: Do not include `https://` in the host name
client = Client(host='bravo.demo.trial.training.cloud', port=9440, user='myUserName', password='myPassWord', secure=True)
```

**Versions**
- ClickHouse Driver: 0.2.6
- ClickHouse Server: 23.10.5.20
- Python: 3.10.0

**Comment by iamtalhaasghar** [comment]
I was also getting this error when moving data from a table on one clickhouse server to another clickhouse server using `insert into remote` function. I changed log levels to `notice` from `trace` and the table was transferred this time witout 
Getting `EOFError: Unexpected EOF while reading byte` error. 

May be I was getting this error due to low RAM space as there are other processes running. When I changed log levels the RAM didn't went beyond 97% and all was ok. https://github.com/mymarilyn/clickhouse-driver/issues/175#issuecomment-1773144393

**Comment by cavalloBrianWerth** [comment]
If anyone is still having issues with this, I also had to set `server_hostname` (constructor param) to the same value as `host_name`
